### ISSUE TITLE
Audio Sample Recording

### ISSUE THREAD

**Issue by netpipe** [LMMS/lmms#1297]
setup a jack sink and get audio into it for recording and saving samples to library. would make audio production a breeze. ( pulseaudio mic support too ? )  could donate 400000 dogecoins for this. email teslaenergy@gmail.com if you want to tackle this one.

**Comment by tamb** [comment]
@diizy is there any progress on audio recording? That would make this program so amazing.

**Comment by tecan** [comment]
another option is to use freelancer.com to buy features. typically 200 for feature and 400 for small application.

**Comment by PhysSong** [comment]
#4994 and #3943 might be related.

**Comment by Reflexe** [comment]
@tamb , @tecan : recording (with jack, sdl and pulseaudio) is pretty much ready. It includes a lot of internal changes so making it stable will take some time. I hope to get it ready to merge in a month and then merge it one month later.

**Comment by tecan** [comment]
Thanks, I would like to tip 100 next month and 100 the month after, hopefully it helps.

**Comment by Reflexe** [comment]
@tecan Thanks! I appreciate it very much.

**Comment by iota-xSK** [comment]
Is there any progress on this? I would love to (try to) help in both cases.

**Comment by Rossmaxx** [comment]
You are the 10 millionth person asking. @LostRobotMusic agreed to get sample recording added once sample caching is implemented. But seems like that PR too is in a limbo.

**Comment by Rossmaxx** [comment]
> could donate 400000 dogecoins for this.

How did we miss this?

**Comment by LostRobotMusic** [comment]
> You are the 10 millionth person asking. @LostRobotMusic agreed to get sample recording added once sample caching is implemented. But seems like that PR too is in a limbo.

I did not.  I said I would add sample editing and extremely high-performance sample drawing.  I never said I would add audio recording.

**Comment by Rossmaxx** [comment]
Ahh i misplaced sample editing and recording.

**Comment by michaelgregorius** [comment]
> > could donate 400000 dogecoins for this.
> 
> How did we miss this?

This would be around 64080 dollars now. :sweat_smile:

**Comment by iota-xSK** [comment]
So is help needed then around sample recording? How difficult of an issue would that be? I am guessing quite difficult but not impossible

**Comment by Rossmaxx** [comment]
Yup we need help. If you have discord, you can join our server to discuss this in detail.

**Comment by Rossmaxx** [comment]
Also we have #5990 but it's incomplete + hard to bring back to master from all the merge conflicts.

**Comment by michaelgregorius** [comment]
I have brought #5990 back from the sleeping with commit 93246581f5b.

There was only one merge conflict which got automatically resolved and I had to adjust the setting of the font size in `SampleClipView::paintEvent`. The setting of the font size was changed with pull request #7185.

I was able to record a guitar clip using the SDL audio driver.

Depending on how active @Reflexe is we should consider to move the branch into the main repository.

It could also be considered to create a project for sample track recording with some necessary tasks like for example making the audio inputs available via the audio engine for the different audio drivers.
